Resources/doc/first_steps.rst

```diff
@@ -0,0 +1,456 @@
+        $dm->persist($product);
+        $dm->flush();
+
+        return new Response('Created product id '.$product->getId());
+    }
+
+.. note::
+
+    If you're following along with this example, you'll need to create a
+    route that points to this action to see it in work.
+
+Let's walk through this example:
+
+* **lines 8-10** In this section, you instantiate and work with the ``$product``
+  object like any other, normal PHP object;
+
+* **line 12** This line fetches Doctrine's *document manager* object, which is
+  responsible for handling the process of persisting and fetching objects
+  to and from MongoDB;
+
+* **line 13** The ``persist()`` method tells Doctrine to "manage" the ``$product``
+  object. This does not actually cause a query to be made to MongoDB (yet);
+
+* **line 14** When the ``flush()`` method is called, Doctrine looks through
+  all of the objects that it's managing to see if they need to be persisted
+  to MongoDB. In this example, the ``$product`` object has not been persisted yet,
+  so the document manager makes a query to MongoDB, which adds a new entry.
+
+If you are using `autowiring`, you can use type hinting to fetch the ``doctrine_mongodb.odm.document_manager`` service:
+
+.. code-block:: php
+
+    // App/Controller/DefaultController.php
+    namespace App\Controller;
+
+    use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+    use Doctrine\ODM\MongoDB\DocumentManager as DocumentManager;
+    use App\Document\Product;
+    use Symfony\Component\HttpFoundation\Response;
+
+    class DefaultController extends AbstractController
+    {
+        public function createProduct(DocumentManager $dm)
+        {
+            $product = new Product();
+            $product->setName('A Foo Bar');
+            $product->setPrice('19.99');
+
+            $dm->persist($product);
+            $dm->flush();
+
+            return new Response('Created product id '.$product->getId());
+        }
+    }
+
+.. note::
+
+    In fact, since Doctrine is aware of all your managed objects, when you
+    call the ``flush()`` method, it calculates an overall changeset and executes
+    the most efficient operation possible.
+
+When creating or updating objects, the workflow is always the same. In the
+next section, you'll see how Doctrine is smart enough to update entries if
+they already exist in MongoDB.
+
+.. tip::
+
+    Doctrine provides a library that allows you to programmatically load testing
+    data into your project (i.e. "fixture data"). For information, see
+    `DoctrineFixturesBundle`_.
+
+Fetching Objects from MongoDB
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Fetching an object back out of MongoDB is even easier. For example, suppose
+you've configured a route to display a specific ``Product`` based on its
+``id`` value:
+
+.. code-block:: php
+
+    public function showAction($id)
+    {
+        $product = $this->get('doctrine_mongodb')
+            ->getRepository('AcmeStoreBundle:Product')

```

Comment:
We should switch to using `::class` constants instead of aliases.

Reply:
Same as previous comment, I'll be fixing these separately